src/components/delivery/Compass.js

```diff
@@ -0,0 +1,73 @@
+import React from "react";
+import { useSelector } from "react-redux";
+import { selectRoverPosition } from "../../store/telemetrySlice";
+import "./Compass.css";
+
+const Compass = () => {
+  const { orientW, orientX, orientY, orientZ } = useSelector(
+    selectRoverPosition
+  );
+
+  // Calculate the direction based on the quaternion values
+  const direction = Math.round(
+    (Math.atan2(
+      2 * (orientW * orientZ + orientX * orientY),
+      1 - 2 * (orientY * orientY + orientZ * orientZ)
+    ) * 180) / Math.PI
+  );
+  const roll = Math.round(
+    (Math.atan2(
+      2 * (orientW * orientX + orientY * orientZ),
+      1 - 2 * (orientX * orientX + orientY * orientY)
+    ) * 180) / Math.PI
+  );
+
+  const pitch = Math.round(
+    2 * Math.atan2(
+      2 * (orientY * orientW - orientX * orientZ),
+      1 - 2 * (orientY * orientY + orientZ * orientZ)
+    ) * (180 / Math.PI)
+  );
+
+  // angle = 2 * acos(c1c2c3 + s1s2s3)
+  // c1 = 1 in all cases, since yaw = 0
+  // s1s2s3 = 0 in all cases, since yaw = 0
+  // math comes from here: https://www.euclideanspace.com/maths/geometry/rotations/conversions/eulerToAngle/index.htm
+
+  const c2 = Math.cos(pitch / 2);
+  const c3 = Math.cos(roll / 2);
```

Comment:
`pitch` and `roll` are calculated above in degrees (they are multiplied by 180/pi) but `Math.cos` requires radians so this will give the wrong answer